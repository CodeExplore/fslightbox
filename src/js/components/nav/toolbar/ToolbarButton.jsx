import React from 'react';
import PropTypes from 'prop-types';
import Svg from "../../helpers/Svg.jsx";

const ToolbarButton = ({ onClick, viewBox, size, d, title }) => (
    <div onClick={ onClick }
         className="fslightbox-toolbar-button fslightbox-flex-centered"
         title={ title }>
        <Svg viewBox={ viewBox } size={ size } d={ d }/>
    </div>
);

ToolbarButton.propTypes = {
    onClick: PropTypes.func.isRequired,
    viewBox: PropTypes.string.isRequired,
    size: PropTypes.string.isRequired,
    d: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
};
export default ToolbarButton;