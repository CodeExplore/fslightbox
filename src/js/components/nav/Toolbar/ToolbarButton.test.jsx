import React from 'react';
import { shallow } from 'enzyme/build';
import ToolbarButton from "../../../../src/components/nav/toolbar/ToolbarButton";

const mockOnClick = jest.fn();
const toolbarButton = shallow(<ToolbarButton
    onClick={ mockOnClick }
    viewBox={ '0 0 20 20' }
    size={ '26px' }
    d={ 'M1425' }
    title={ 'title' }
/>);

describe('ToolbarButton DOM', () => {
    it('should match snapshot', () => {
        expect(toolbarButton).toMatchSnapshot();
    });
});

describe('calling onClick received from props', () => {
    beforeEach(() => {
        toolbarButton.simulate('click');
    });

    it('should call ', () => {
        expect(mockOnClick).toBeCalled();
    });
});