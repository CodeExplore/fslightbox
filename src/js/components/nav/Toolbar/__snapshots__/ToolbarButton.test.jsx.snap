// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToolbarButton DOM should match snapshot 1`] = `
<div
  className="fslightbox-toolbar-button fslightbox-flex-centered"
  onClick={[MockFunction]}
  title="title"
>
  <Svg
    d="M1425"
    size="26px"
    viewBox="0 0 20 20"
  />
</div>
`;
